plugins/woocommerce/src/Internal/Admin/Marketing/MarketingSpecs.php

```diff
@@ -136,43 +136,29 @@ protected function is_marketing_channel_plugin( array $plugin_data ): bool {
 	/**
 	 * Load knowledge base posts from WooCommerce.com
 	 *
-	 * @param string|null $term Term of posts to retrieve.
+	 * @param string|null $topic The topic of marketing knowledgebase to retrieve.
 	 * @return array
 	 */
-	public function get_knowledge_base_posts( ?string $term ): array {
-		$terms = array(
-			'marketing' => array(
-				'taxonomy' => 'category',
-				'term_id'  => 1744,
-				'argument' => 'categories',
-			),
-			'coupons'   => array(
-				'taxonomy' => 'post_tag',
-				'term_id'  => 1377,
-				'argument' => 'tags',
-			),
-		);
+	public function get_knowledge_base_posts( ?string $topic ): array {
+		$available_topics = array( 'marketing', 'coupons' );
```

Comment:
❔ I'm thinking we don't need this `$available_topics` array. Having it means that we would need to maintain the list here in this function.

Imagine a scenario where in the future we want a new page with a new topic called `'abc'`. In the new page, we would need to make a call to `get_knowledge_base_posts('abc')`. 

With this `$available_topics` array, we would need to add `'abc'` into the array, otherwise it would default to `'marketing'`. 

Without this `$available_topics` array, we don't need to do anything in this function, and it would just work.

Reply:
Thank you, @ecgan, for raising this question. I agree that the `$available_topics` is redundant. However, I was trying to preserve the existing condition to display `marketing` posts when no other topic is set other than `coupons`. Let me know what you think.

Reply:
> However, I was trying to preserve the existing condition to display `marketing` posts when no other topic is set other than `coupons`

Thanks for explaining that. I just realized that behavior, but I'm also thinking that existing behavior may not be a desirable experience, in my opinion.

Imagine the following scenarios:

```php
get_knowledge_base_posts('marketing') 
// The caller expects this to return marketing posts. Works as expected.

get_knowledge_base_posts('coupons') 
// The caller expects this to return coupons posts. Works as expected.

get_knowledge_base_posts('abc') 
// The caller expects this to return "abc" posts, but it returns marketing posts. 
// The caller / developer may not realize it, and end up displaying wrong posts in the UI.
```

Also, if we really want to have the existing behavior, I'm thinking it would be better to have that logic in the WCCOM API. 

I looked into your WCCOM API PR `get_items` function https://github.com/Automattic/woocommerce.com/pull/18111/files#diff-914ff9b3d818c5698700ec1267640836db8f27b5d57cea69c9dd643332da7ac2R103-R111 and I thought it would give the existing behavior, but I think the enum https://github.com/Automattic/woocommerce.com/pull/18111/files#diff-914ff9b3d818c5698700ec1267640836db8f27b5d57cea69c9dd643332da7ac2R87 is preventing that, so when I make a `GET https://woocommerce.com/wp-json/wccom/marketing-knowledgebase/v1/posts/abc` request, it responds with HTTP 400 and says "topic is not one of marketing and coupons".

I think it is better to have this developer experience, which WCCOM API already supports:

```php
get_knowledge_base_posts('abc') 
// The caller expects this to return "abc" posts, but it returns error. 
// The caller / developer should notice this issue easily, and take further action, e.g. handling the 'abc' category in WCCOM API.
```

What do you think? 🤔

Reply:
This has been addressed in [`01b99e9` (#40090)](https://github.com/woocommerce/woocommerce/pull/40090/commits/01b99e98036e365d7e57112f2c539b12789aac7f). 👍